### PR TITLE
New knight

### PR DESCRIPTION
**Pull request by ghost**
- Added logic for Knight

### CODE REVIEW

File: app/models/pieces/knight.rb [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
-end
\ No newline at end of file
+
+  def valid_move?(x, y)
+    # Calculate old move and new move
+    x_move = (x_position - x).abs
+    y_move = (y_position - y).abs
+
+    # If x or y move is 2 then the next x or y move is 1
+    if x_move == 2 || y_move == 2
```
Comment (ilyakrasnov) [inline app/models/pieces/knight.rb]: Seems valid.. Can be refactored a little bit though:

you basically want to check if x_move, y_move == 2 and 1 or 1 and 2... you could add a method like:

def valid_move_combination?(x_move, y_move)
(x_move == 2 && y_move == 1) || (x_move == 1 && y_move ==2) ? true : false
end

then in valid_move? you define the x_move and y_move and call the new method..

File: spec/models/knight_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+require 'knight'
+
+RSpec.describe Knight, type: :model do
+
+  before do
+    # Create test data example
+    @knight = Knight.create(:x_position => 3, :y_position => 3)
+  end
+
+  it "should know if x or y move is two then the next x or y move is one vice versa" do
```
Comment (ilyakrasnov) [inline spec/models/knight_spec.rb]: Probably no need for this if you refactor as above.

Try to test the overall logic more - place you piece on different positions and try to move to different places and see if you get true or false..

File: spec/models/knight_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    x_bottom_x = (@knight.x_position - x_bottom_x).abs # => 2
+    x_left_y = (@knight.y_position - x_left_y).abs # => 1
+    x_right_y = (@knight.y_position - x_right_y).abs # => 1
+
+    y_top_x = (@knight.x_position - y_top_x).abs # => 1
+    y_bottom_x = (@knight.x_position - y_bottom_x).abs # => 1
+    y_left_y = (@knight.y_position - y_left_y).abs # => 2
+    y_right_y = (@knight.y_position - y_right_y).abs # => 2
+
+    expect(x_top_x).to be == 2
```
Comment (ilyakrasnov) [inline spec/models/knight_spec.rb]: Like mentioned above, you test here if ruby can subtract correctly :) we would need more logic test of our valid move..

File: app/models/pieces/knight.rb (hunk 1 above)
Comment (ghost) [inline app/models/pieces/knight.rb]: Thanks. That didn't even cross my mind.

**Pull request closed without merging**